File: FrontEnd/lib/domain/interfaces/usecases/user/i_get_user_by_id_use_case.dart
/// @file i_get_user_by_id_use_case.dart
/// @description Interfaz: Obtiene el perfil público de un usuario por ID.
/// @module User
/// @layer Domain
library;

import '../../../../domain/entities/user.dart';

abstract interface class IGetUserByIdUseCase {
  /// Obtiene el perfil público de un usuario por ID.
  ///
  /// [throws] AppError.notFound si el usuario no existe.
  Future<User> execute(String userId);
}

File: BackEnd/src/domain/interfaces/usecases/user/IGetUserByIdUseCase.ts
/**
 * @file IGetUserByIdUseCase.ts
 * @description Interfaz del caso de uso para obtener el perfil de un usuario por ID.
 * @module User
 * @layer Domain
 */

import type { UserResponseDTO } from '../../../dtos/user/user-response.dto.js';

export interface IGetUserByIdUseCase {
  /**
   * @param userId — ID del usuario a obtener.
   * @returns Datos públicos del usuario.
   * @throws {NotFoundException} Si el usuario no existe.
   */
  execute(userId: string): Promise<UserResponseDTO>;
}
